client/task-list/tasks.js

```diff
@@ -39,6 +39,17 @@ export function recordTaskViewEvent(
 	} );
 }
 
+const groupBy = ( array, key ) => {
```

Comment:
Ideally we should have a basic set of tests for this. Also it's a very generic collections method that we should share under lib or something.

Reply:
Good idea! I moved the method to `lib/lists` and added some tests in the commit 600651c